feat(oidc): return defined error when discovery failed

ay4toh5i opened the pull request:
This PR adds a specific error to handle discovery failures. 
It allows proper error handling when the issuer URL is invalid or the configuration is missing.
Thank you.

### Definition of Ready
- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

muhlemmer commented on pkg/client/client_test.go:
```diff
@@ -32,16 +33,27 @@ func TestDiscover(t *testing.T) {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := Discover(context.Background(), tt.args.issuer, http.DefaultClient, tt.args.wellKnownUrl...)
-			if tt.wantErr {
-				assert.Error(t, err)
+			if tt.wantErr != nil {
+				assert.ErrorIs(t, err, tt.wantErr)
 				return
 			}
+
 			require.NoError(t, err)
 			if tt.wantFields == nil {
```
Error check be simplefied:

```suggestion
			got, err := Discover(context.Background(), tt.args.issuer, http.DefaultClient, tt.args.wellKnownUrl...)
			require.ErrorIs(t, err, tt.wantErr)
```

muhlemmer commented on pkg/client/client_test.go:
```diff
@@ -32,16 +33,27 @@ func TestDiscover(t *testing.T) {
 			got, err := Discover(context.Background(), tt.args.issuer, http.DefaultClient, tt.args.wellKnownUrl...)
-			if tt.wantErr {
-				assert.Error(t, err)
+			if tt.wantErr != nil {
+				assert.ErrorIs(t, err, tt.wantErr)
 				return
 			}
+
 			require.NoError(t, err)
```
Error check can be simplefied:

```suggestion
            require.ErrorIs(t, err, tt.wantErr)
```

ay4toh5i replied:
I understand the suggestion to simplify the error check, but the current implementation uses early returns when an error is encountered to skip the remaining tests. If we simplify the check as proposed, it may cause the tests to fail.

muhlemmer replied:
When the expectation in `wantFields` is correct (`nil` in this case) the test will not fail.

ay4toh5i replied:
It seems I initially misunderstood your suggestion, but after reconsidering, I’ve incorporated your proposed changes. Thanks!

ay4toh5i commented:
@muhlemmer 
Thanks for taking the time to review my PR and for the feedback! 
I’ve addressed the points you mentioned. Could you take another look when you get a chance?
